File: src/main/java/com/hsb/crm/entity/Visit.java
package com.hsb.crm.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import java.time.LocalDateTime;
import java.io.Serializable;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

/**
 * <p>
 * 
 * </p>
 *
 * @author hsb
 * @since 2022-07-18
 */
@Data
@EqualsAndHashCode(callSuper = false)
@Accessors(chain = true)
public class Visit implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * 拜访编号
     */
    @TableId(value = "vid", type = IdType.AUTO)
    private Integer vid;

    /**
     * 所属公司
     */
    private String company;

    /**
     * 拜访客户
     */
    private String client;

    /**
     * 联系电话
     */
    private String phone;

    /**
     * 陪同人员
     */
    private String accompany;

    /**
     * 拜访主题
     */
    private String theme;

    /**
     * 拜访目的
     */
    private String purpose;

    /**
     * 拜访前准备
     */
    private String ready;

    /**
     * 拜访结果
     */
    private String outcome;

    /**
     * 客户态度
     */
    private String attitude;

    /**
     * 活动经费
     */
    private String fund;

    /**
     * 备注
     */
    private String remark;

    /**
     * 记录人id
     */
    private Integer operid;

    /**
     * 拜访时间
     */
    private LocalDateTime createtime;


}

File: services/visit.js
import {VISITADD} from '@/services/api'
import {LISTVISIT,DELETEVISIT} from '@/services/api'
import {request, METHOD} from '@/utils/request'
/**
 * 添加角色服务
 * @param name 账户名
 * @param password 账户密码
 * @returns {Promise<AxiosResponse<T>>}
 */
//添加拜访记录
async function addvisit(visit) {
  return request(VISITADD, METHOD.POST, {
    company: visit.company,
    client:visit.client,
    phone:visit.phone,
    accompany:visit.accompany,
    theme:visit.theme,
    purpose:visit.purpose,
    ready:visit.ready,
    outcome:visit.outcome,
    attitude:visit.attitude,
    fund:visit.fund,
    remark:visit.remark,
    createtime:visit.createtime
  })
}
//查询拜访记录
async function listvisit(current,size) {
return request(LISTVISIT, METHOD.GET, {
  current:current,
  size:size
})
}
//记录者本人删除拜访记录
async function deletevisit(vid) {
    return request(DELETEVISIT, METHOD.POST, {
        vid:vid
    })
    }
export { addvisit,listvisit,deletevisit}